packages/pwa/webpack.config.js

```diff
@@ -66,13 +68,95 @@ module.exports = {
             PL_TERMS_OF_SERVICE: terms_of_service,
         }),
         new CleanWebpackPlugin(),
+        {
+            apply(compiler) {
+                compiler.hooks.compilation.tap("Store Built Files for CSP", (compilation) => {
+                    HtmlWebpackPlugin.getHooks(compilation).beforeEmit.tapAsync(
+                        "Store Built Files for CSP",
+                        (data, callback) => {
+                            const isBuildingLocally = pwaUrl.startsWith("http://localhost");
+                            const fileExtensionsToCspRule = new Map([
+                                ["js", "script-src"],
+                                ["map", "script-src"],
+                                ["woff2", "font-src"],
+                                ["svg", "img-src"],
+                                ["png", "img-src"],
+                            ]);
+                            const builtFilesForCsp = new Map([
+                                ["script-src", []],
+                                ["font-src", []],
+                                [
+                                    "img-src",
+                                    [
+                                        "favicon.png",
+                                        // TODO: These should be dynamically added (from the manifest), but it's not available at this point
+                                        "icon_512x512.e3175643e8fe0d95175a493da5201480.png",
+                                        "icon_384x384.971e45062e4d601a3014dc16ee3ed27b.png",
+                                        "icon_256x256.9a47fba2857d94939047064f37cd075f.png",
+                                        "icon_192x192.8dfb7236c7e6b6591567173b18eaa144.png",
+                                        "icon_128x128.f620784d1682c9fbb033d3b018e7d998.png",
+                                        "icon_96x96.eda9f98be1c35dabab77f9d2ab7be538.png",
+                                    ],
+                                ],
+                                // TODO: This should to be dynamically added, but it's not available at this point
+                                ["manifest-src", ["manifest.623e2268f17398ec7f19225e281e4056.json"]],
```

Comment:
Have you tried simply adding your plugin **after** all the other plugins?

Reply:
Good idea. My only bet would be to use the `afterEmit` hook, since the manifest and inject ones use `emit`, _but_ then I'd be stuck having to read the file and write it, while some other plugin might also try to do that. Just very error-prone stuff.

Reply:
I think we should still try and find a solution for this, since these files **will** change sooner or later, plus anyone trying to customise anything in `assets` will run into problems here. I'm ok with deferring this until later, though. If we do so, let's create an issue for this.

Reply:
Deferred. Issue is at https://github.com/padloc/padloc/issues/471